### ISSUE TITLE
SqlSmith: Fix operators with NumericOutOfRange

### ISSUE THREAD

**Issue by kwannoel** [singularity-data/risingwave#3900]
```sql
SELECT (SMALLINT '22' << SMALLINT '34') / 21.81402348310204 AS col_0, REAL '18.92711622257903' AS col_1;
-- shrunk
SELECT (SMALLINT '22' << SMALLINT '34');
-- runtime, where v:= '300'
SELECT (SMALLINT '22' << SMALLINT v);
```

```sh
thread 'main' panicked at 'Failed to execute query: db error: ERROR: RPC error: Status { code: Internal, message: "Expr error: NumericOutOfRange", metadata: MetadataMap { headers: {"risingwave-error-bin": "CBwSHUV4cHIgZXJyb3I6IE51bWVyaWNPdXRPZlJhbmdl"} }, source: None }
```

**Comment by kwannoel** [comment]
I think the right way to handle this is to permit `NumericErrors`. Since these are runtime errors from **valid queries**.

**Comment by kwannoel** [comment]
The alternative is to be conservative about the types of expressions we generate.
For instance when generating lshift / rshift / div, we constraint the left and right exprs to set of valid expressions.
In the above case, it means only literals, or compound expressions of literals. Additionally these should have valid bounds (e.g. right expr of div is non-zero).
We can't have column exprs, since these can still fail during runtime.

**Comment by kwannoel** [comment]
Note that the erroring behaviour here diverges from postgres, this is intentional: https://github.com/singularity-data/risingwave/blob/b4eb1107bc16f8d583563f776f748632ddcaa0cb/src/expr/src/vector_op/bitwise_op.rs#L24. 

We see similar behaviours for other arithmetic expressions e.g. division by zero.

**Comment by kwannoel** [comment]
Resolved by #3929
